.github/workflows/ci.yml

```diff
@@ -49,16 +49,9 @@ jobs:
           - 17
     timeout-minutes: 45
     steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0 # checkout all commits, as the build result depends on `git describe` equivalent
-      - uses: actions/setup-java@v3
+      - uses: trinodb/github-actions/checkout-and-setup@6a40cfbfa0ad484d69f5482b29c41fbaac72944e
```

Comment:
must live inside the repo instead of being versioned outside - easy to test and easy to see what is actually being done.

Reply:
the checkout is a one-liner anyway so can be done before the composite action and composite action can do other heavy lifting.

Reply:
It's not a one-liner:
```
      - uses: actions/checkout@v3
        with:
          fetch-depth: 0 # checkout all commits, as the build result depends on `git describe` equivalent
```
Also, for #12817 we will need to verify if the checked-out commit is the one that's allowed, so we'll need an external action anyway. I'd rather avoid the additional work of creating a local action here and then removing it. I have an impression that we want #12817 to get merged soon, but I thought we could get these changes merged even sooner, as it simplifies the workflow and makes it easier to maintain, especially in forks.

Note that #12817 is open for 5 months, I keep rebasing it on a regular basis but never had to update the action (it exists for 4 months). I don't expect it to change that often, so I'm not sure there's a lot to gain if we'd split it up just to keep one definition in this repo.

Reply:
I moved the action here and left the checkout step.